include/SFML/Audio/Sound.hpp

```diff
@@ -213,18 +207,27 @@ class SFML_AUDIO_API Sound : public SoundSource
     ////////////////////////////////////////////////////////////
     Sound& operator=(const Sound& right);
 
+private:
+    friend class SoundBuffer;
```

Comment:
This is needed because..?

Reply:
When the SoundBuffer updates its OpenAL buffer, we need first to dettach sounds currently bound to it to avoid OpenAL errors. Previously we had `Sound::resetBuffer` method for that same reason. That method was public but documented as "for internal use only". I replaced it with methods `Sound::detachBuffer` and `Sound::reattachBuffer` and figured it would be better to make them private and having friend class SoundBuffer because the methods are only useful to SoundBuffer.